src/components/TodoCard.jsx

```diff
@@ -0,0 +1,37 @@
+import { useDispatch } from "react-redux";
+import { achieveTodo, deleteTodo, updateTask } from "../features/todo";
+import { FaTrash } from "react-icons/fa";
+
+export default function TodoCard({ todo }) {
+  const dispatch = useDispatch();
+
+  return (
+    <div className="flex justify-between w-full items-center gap-3 border-b border-orange-200 py-2">
+      <div className="flex w-[80%] items-center gap-5">
+        <input
+          type="checkbox"
+          checked={todo.done}
+          onChange={() => dispatch(achieveTodo(todo))}
+        />
+        <p
+          contentEditable={true}
+          suppressContentEditableWarning={true}
+          onBlur={(e) =>
+            dispatch(updateTask({ ...todo, task: e.target.textContent }))
```

Comment:
I would avoid spreading this todo object, and just pass in. two parameters one could be the old todo, and the second would be the change object.